core/commands/add.go

```diff
@@ -29,6 +31,7 @@ const progressReaderIncrement = 1024 * 256
 const (
 	progressOptionName = "progress"
 	wrapOptionName     = "wrap-with-directory"
+	hiddenOptionName   = "hidden"
```

Comment:
I am not very found of talking about "hidden" files. It think it is confusing to talk about both "ignored" and "hidden" files.

If we want some files or directories to be ignored by default, then `ipfs init` should create a default `.ipfsignore` file that contains lines to ignore those files, though it is true that there may be a few hardcoded files or directories that are always ignored (for example in Git the `.git` directory is obviously always ignored).

By the way it is good anyway if `ipfs init` creates `.ipfsignore` even if it is empty, but it's even better if it is created with some comments that explain how it works.

By default ignored files should not be added by `ipfs add` except if a special option is passed. In Git you must use `git add --force` to add ignored files, so I would suggest `--force` too or maybe `--force-ignored`.